File: sdk/typescript/src/agents.ts
/**
 * TypeScript SDK Client for Enterprise AI Agent Orchestration Platform (Phase 5.36).
 */

export class AgentsClient {
  private baseUrl: string;
  private apiKey?: string;

  constructor(baseUrl: string, apiKey?: string) {
    this.baseUrl = baseUrl;
    this.apiKey = apiKey;
  }

  private getHeaders(): Record<string, string> {
    const headers: Record<string, string> = {
      'Content-Type': 'application/json',
    };
    if (this.apiKey) {
      headers['Authorization'] = `Bearer ${this.apiKey}`;
    }
    return headers;
  }

  async register(tenantId: string, payload: { name: string; agent_type?: string; role?: string; capabilities?: string[]; description?: string }): Promise<any> {
    const res = await fetch(`${this.baseUrl}/v1/agents/register?tenant_id=${tenantId}`, {
      method: 'POST',
      headers: this.getHeaders(),
      body: JSON.stringify(payload),
    });
    if (!res.ok) throw new Error(`Agent register failed: ${res.statusText}`);
    return res.json();
  }

  async list(tenantId: string = 'default'): Promise<any> {
    const res = await fetch(`${this.baseUrl}/v1/agents?tenant_id=${tenantId}`, {
      headers: this.getHeaders(),
    });
    if (!res.ok) throw new Error(`Agent list failed: ${res.statusText}`);
    return res.json();
  }

  async get(agentId: string, tenantId: string = 'default'): Promise<any> {
    const res = await fetch(`${this.baseUrl}/v1/agents/${agentId}?tenant_id=${tenantId}`, {
      headers: this.getHeaders(),
    });
    if (!res.ok) throw new Error(`Agent get failed: ${res.statusText}`);
    return res.json();
  }

  async executeTask(tenantId: string, payload: { goal: string; prompt: string; agent_id?: string; is_destructive?: boolean }): Promise<any> {
    const res = await fetch(`${this.baseUrl}/v1/agents/execute?tenant_id=${tenantId}`, {
      method: 'POST',
      headers: this.getHeaders(),
      body: JSON.stringify(payload),
    });
    if (!res.ok) throw new Error(`Task execute failed: ${res.statusText}`);
    return res.json();
  }

  async getTrace(traceId: string, tenantId: string = 'default'): Promise<any> {
    const res = await fetch(`${this.baseUrl}/v1/agents/traces/${traceId}?tenant_id=${tenantId}`, {
      headers: this.getHeaders(),
    });
    if (!res.ok) throw new Error(`Trace get failed: ${res.statusText}`);
    return res.json();
  }

  async getAnalytics(tenantId: string = 'default'): Promise<any> {
    const res = await fetch(`${this.baseUrl}/v1/agents/analytics?tenant_id=${tenantId}`, {
      headers: this.getHeaders(),
    });
    if (!res.ok) throw new Error(`Analytics get failed: ${res.statusText}`);
    return res.json();
  }
}

File: sdk/go/agents.go
package llmengine

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"
	"net/http"
)

type AgentsClient struct {
	client  *http.Client
	baseURL string
	apiKey  string
}

func NewAgentsClient(baseURL string, apiKey string) *AgentsClient {
	return &AgentsClient{
		client:  &http.Client{},
		baseURL: baseURL,
		apiKey:  apiKey,
	}
}

func (c *AgentsClient) RegisterAgent(tenantID string, name string, agentType string, role string) (map[string]interface{}, error) {
	payload := map[string]interface{}{
		"name":       name,
		"agent_type": agentType,
		"role":       role,
	}
	body, err := json.Marshal(payload)
	if err != nil {
		return nil, err
	}

	req, err := http.NewRequest("POST", fmt.Sprintf("%s/v1/agents/register?tenant_id=%s", c.baseURL, tenantID), bytes.NewBuffer(body))
	if err != nil {
		return nil, err
	}
	req.Header.Set("Content-Type", "application/json")
	if c.apiKey != "" {
		req.Header.Set("Authorization", "Bearer "+c.apiKey)
	}

	resp, err := c.client.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	respBody, err := io.ReadAll(resp.Body)
	if err != nil {
		return nil, err
	}

	var res map[string]interface{}
	err = json.Unmarshal(respBody, &res)
	return res, err
}

func (c *AgentsClient) ExecuteTask(tenantID string, goal string, prompt string, agentID string) (map[string]interface{}, error) {
	payload := map[string]interface{}{
		"goal":     goal,
		"prompt":   prompt,
		"agent_id": agentID,
	}
	body, err := json.Marshal(payload)
	if err != nil {
		return nil, err
	}

	req, err := http.NewRequest("POST", fmt.Sprintf("%s/v1/agents/execute?tenant_id=%s", c.baseURL, tenantID), bytes.NewBuffer(body))
	if err != nil {
		return nil, err
	}
	req.Header.Set("Content-Type", "application/json")
	if c.apiKey != "" {
		req.Header.Set("Authorization", "Bearer "+c.apiKey)
	}

	resp, err := c.client.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	respBody, err := io.ReadAll(resp.Body)
	if err != nil {
		return nil, err
	}

	var res map[string]interface{}
	err = json.Unmarshal(respBody, &res)
	return res, err
}
